lib/reek/smells/smell_repository.rb

```diff
@@ -9,29 +9,37 @@ module Smells
     #
     # @api private
     class SmellRepository
-      attr_reader :detectors
+      attr_reader :detectors, :typed_detectors, :configuration, :source_description, :smell_types
 
       def self.smell_types
         Reek::Smells::SmellDetector.descendants.sort_by(&:name)
       end
 
-      def initialize(source_description = nil, smell_types = self.class.smell_types)
-        @typed_detectors = nil
-        @detectors = {}
-        smell_types.each do |klass|
-          @detectors[klass] = klass.new(source_description)
+      def initialize(source_description: nil,
+                     smell_types: self.class.smell_types,
+                     configuration: Configuration::AppConfiguration.new)
+        self.source_description = source_description
+        self.typed_detectors    = nil
+        self.detectors          = {}
+        self.configuration      = configuration
+        self.smell_types        = smell_types
```

Comment:
Not directly part of this PR but according to the boyscout rule I also reduced the usage of instance variables and preferred proper setters.